### PR TITLE
First phase of number to integer conversion

### LINKED ISSUE #3214: sntp module uses lua_pushnumber when it should use lua_pushinteger.
Current behavior is broken in the lua53 build.

### PR DESCRIPTION
**Pull request by pjsg**
Fixes #3214, #3215.

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [ ] I have thoroughly tested my contribution.
- [ ] The code changes are reflected in the documentation at `docs/*`.

Changing calls to `checknumber` and `pushnumber` to be `checkinteger` and `pushinteger` as this is (nearly always) the right thing to do.

### CODE REVIEW

**Comment by nwf** [comment]
These all look fine.

**Comment by TerryE** [comment]
@pjsg, I've gone through the rest of the updates and done another push.  There's a conflict in `app/modules/sjson.c` but I'd like you to review this anyway so you may as well fix this via the web editor.  Thankis Terry

**Comment by pjsg** [comment]
I've resolved the conflict in sjson.c -- it builds with both 5.1 and 5.3 versions. But no testing beyond that. I'll merge this into my iot device build and give it a whirl.

**Comment by TerryE** [comment]
Thanks Philip. Like you, all I did was a full Travis clean compile on both versions. I don't have the H/W to test a lot of these, but the changes are mainly number to integer swaps when the C-side variables are `int`s anyway.

**Review by TerryE — APPROVED** [review]

**Comment by TerryE** [comment]
@nwf Nathaniel, perhaps you could do a review of these and merge them into `dev`.  Thanks

File: app/modules/node.c [hunk 1]
```diff
@@ -76,34 +77,27 @@ static int node_deepsleep( lua_State* L )
...
-  //us = luaL_checkinteger( L, 1 );
   // Set deleep option, skip if nil
   if ( lua_isnumber(L, 2) )
   {
-    option = lua_tointeger(L, 2);
-    if ( option < 0 || option > 4)
-      return luaL_error( L, "wrong arg range" );
-    else
-      system_deep_sleep_set_option( option );
+    luaL_argcheck(L, 2, (option = lua_tounsigned(L, 2)) <= 4, "wrong option value" );
```
Comment (nwf) [inline app/modules/node.c:83]: Yick!  I understand that that's going to compile to the right thing, but please just use another line to assign option before the check.

**Review by nwf — CHANGES_REQUESTED** [review]

File: app/modules/node.c (hunk 1 above)
Comment (TerryE) [reply app/modules/node.c:83]: Who is the boff around here? 🤣
Maybe you want to rewrite all the nasty Lua macros as well?

Comment (nwf) [reply app/modules/node.c:83]: I just mean to say that I think
```
option = lua_tounsigned(L, 2);
luaL_argcheck(L, 2, option <= 4, "wrong option value" );
```
looks better.

Comment (HHHartmann) [reply app/modules/node.c:83]: So do I

**Comment by TerryE** [comment]
Just winding you up 😉. Make the change.

File: app/modules/node.c (hunk 1 above)
Comment (pjsg) [reply app/modules/node.c]: Done

File: app/modules/rtcmem.c [hunk 2]
```diff
@@ -6,10 +6,10 @@
 
 static int rtcmem_read32 (lua_State *L)
 {
-  int idx = luaL_checknumber (L, 1);
+  int idx = luaL_checkinteger (L, 1);
```
[the remark this answers is not in the document]
Comment (pjsg) [reply app/modules/rtcmem.c:9]: I hope that nobody expects that `rtcmem.read32(2.5)` actually does something meaningful!

File: app/modules/rotary.c [hunk 3]
```diff
@@ -254,8 +254,8 @@ static int lrotary_getpos( lua_State* L )
     return 0;
   }
 
-  lua_pushnumber(L, (pos << 1) >> 1);
-  lua_pushnumber(L, (pos & 0x80000000) ? MASK(PRESS) : MASK(RELEASE));
+  lua_pushinteger(L, (pos << 1) >> 1);
+  lua_pushinteger(L, (pos & 0x80000000) ? MASK(PRESS) : MASK(RELEASE));
 
   return 2;
```
[the remark this answers is not in the document]
Comment (pjsg) [reply app/modules/rotary.c:260]: Fixed

**Comment by TerryE** [comment]
Phil's last change fixed @nwf 's niggle so I reckon that we are good to go on this.

**Pull request merged**